### DISCUSSION TITLE
2008/07/15/%E5%BC%80%E5%8F%91-xcontrol

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#232] [Announcements]
# 2008/07/15/%E5%BC%80%E5%8F%91-xcontrol

最新内容查看： https://lv.qizhen.xyz

https://blog.qizhen.xyz/2008/07/15/%E5%BC%80%E5%8F%91-xcontrol

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2009年09月16日 11:16 编辑
目前的控件已经满足用户普遍需求了。对于客户特殊化的需求，最好是由用户自定义完成。关于Picture控件，我觉得画点和线已经是最基本的操作了。公司内部的人使用这个控件，也就是调用它提供出来的这几个VI。Picture内部的实现，可能需要调用系统API提供的画图函数，但具体怎么实现的我也不是太了解。使用它的时候，其实也没必要了解它的内部格式。我从它的机构上猜测，Picture控件实际上记录的是用户的画图过程，它有自己的记录格式。基本格式是flatten后的簇数据。然后再内部调用系统的绘图函数，把图形绘制出来。当然，我没研究过这部分内容，只是自己猜测一下。

回复
彬
2009年09月15日 21:51 编辑
阮大侠： 对于labVIEW常用控件库从labVIEW7.0到LabVIEW2009都似乎没什么变化，而通过我的观察，控件库中添加最多的可以说是应用Picture控件自定义Xcontrol的3D控件了，由此趋势也可以发现Picture控件其实是最好用来自定义特殊界面的控件了，虽然labVIEW中开放了诸如：画点，画线等基本函数用来操作Picture控件画图，但还是不如直接深入工具包的内核探究其中如何操作Picture控件的。通过底层的代码我大概了解到Picture的一些特点，但只能管中窥豹，觉得学的很辛苦，请问是否能提供一些资料（Picture中的定义，格式）？

回复

WL
2008年07月16日 09:02 编辑
哎不懂

回复

死了都要C
2008年09月13日 10:38 编辑
阮大侠：
  小弟新用LV，目前遇到一些难题，希望阮大侠能帮组解决。  
   1.程序运行速度的问题。目前做一个电路板的测试，测试项有很多，40多个大项，大项中有30多个小项，
     程序的运行是采用动态调用实现的，刚开机运行速度还可以，但过一会非常慢，
     是否与动态调用的资源释放有关系？
   2.6608板卡测量两个信号跳变的滞后时间，
      基本原理是用LV例程中的TWO EDGE SEPERATION MEASURENT.VI来实现的，硬件上采用先到信号连至AUX端，后发信号连至GATE端。
      测量两个信号上升沿跳变的滞后时间是可行的，但测下降沿时始终是0，实在不知原因为何。
  3.6608测量两个信号跳变的滞后时间的时机问题。
     由于要测的这两个脚（PIN100.PIN101）信号的跳变是因另外一脚(PIN99)加高电平引起的，而这两个信号的跳变滞后时间非常短1MS左右，99脚加电平也是通过一数字板卡实现的（PXI6120），目前采取的方案是测量时间的VI跟99脚加电平并行执行，即将这两个VI的ERROR IN 并联。
      请教阮大侠有没有更好的方案。
       不胜感激。

回复

WL
2008年08月4日 09:56 编辑
专业啊

回复

Boone
2008年08月23日 22:26 编辑
我代表LabVIEW_Home，欢迎你来指导！
QQ群：13267790
Boone

回复
Qizhen
2008年08月5日 18:07 编辑
我后来没有吃任何药。只是加强锻炼，注意休息。症状缓解多了，现在只有偶尔才发病。

回复
文治
2008年08月5日 13:17 编辑
看了你的博客,我也有划痕症,请问你的好了没啊,怎么治的啊?
能加我QQ22623658么?或者邮件:liwenzhi84@163.com
不胜感谢

回复
